### ISSUE TITLE
Exposure PID controller

### ISSUE THREAD

**Issue by tschmidt64-SC** [AlexisTM/aero-optical-flow-ros#3]
We were noticing odd exposure behavior with our camera (rapid flickering between high and low exposure) so I decided to look at the source to see what was going on. 

I think there might be some mistakes in this code. It seems that the gain starts at 127 and never changes, so it only ever enters the if condition (because of the `if (gain > 1.0 || ...)`), and never the else condition. Even more strangely, despite the fact that it never enters the else condition (which is the part that modifies the exposure), the exposure continues goes up until it overflows uint16_t. Also, you're assigning float literals to a uint16_t data types, which I also don't understand. Is this code correct? 

https://github.com/AlexisTM/aero-optical-flow-ros/blob/9b71d444d2e8fa66bf02a88d7457594cb0ef34b8/src/mainloop.cpp#L223-L275

**Comment by AlexisTM** [comment]
Hi, for some reason, I do not have the issue but I don't deny there could be bugs. 

The full optical flow and camera control is an exact copy of Intel's original optical flow service, I just added a way to visualise the image from ros and use it for tag tracking.

To be frank, I did not had time to review the optical flow code as I am more thinking of making a nodelet for the camera and a nodelet for the optical flow to separate the logic before really using the node. 

My first guess would be that the integrator sum is not limited. If it's the case, it could create oscillations by overflow. 

Feel free to create a PR :)

**Comment by tschmidt64-SC** [comment]
Thanks for the response! We will be looking into this more today and we'll let you know what we find. 

Quick question: were the #defines at the top of the file there when you started modifying it? I'm curious where these numbers came from. I understand if you didn't touch those and aren't aware, just thought I would ask.

For example, EXPOSURE_ABOSULUTE_MAX_VALUE 1727

Is this just a hardware constraint on the max exposure?

**Comment by AlexisTM** [comment]
Indeed, the defines were there originally. I changed the default resolution. Note also that having a different resolution should make the hardcoded calibration outdated.

Ideally, it should use the calibration provided by ROS, which is now available.

Base software:

https://github.com/intel-aero/aero-optical-flow

This define looks like an experimental number.
